tests/testthat/adtteSpec/tests/testthat/test-shinytest2.R

```diff
@@ -0,0 +1,51 @@
+library(shinytest2)
+
+test_that("{shinytest2} recording: adtteSpecServer module works as expected in the test app", {
+  app <- AppDriver$new(
+    name = "adtteSpecServer module works as expected in the test app",
+    height = 520, width = 979
+  )
+  app$set_inputs(`teal-main_ui-filter_panel-add_ADTTE_filter-filter-var_to_add` = character(0))
+  app$set_inputs(`teal-main_ui-filter_panel-add_MAE_filter-subjects-var_to_add` = character(0))
+  app$set_inputs(`teal-main_ui-filter_panel-add_MAE_filter-hd1-row_to_add` = character(0))
+  app$set_inputs(`teal-main_ui-filter_panel-add_MAE_filter-hd1-col_to_add` = character(0))
+  app$set_inputs(`teal-main_ui-filter_panel-add_MAE_filter-hd2-row_to_add` = character(0))
+  app$set_inputs(`teal-main_ui-filter_panel-add_MAE_filter-hd2-col_to_add` = character(0))
+  app$set_inputs(`teal-main_ui-filter_panel-add_MAE_filter-hd3-row_to_add` = character(0))
+  app$set_inputs(`teal-main_ui-filter_panel-add_MAE_filter-hd3-col_to_add` = character(0))
+  app$set_inputs(`teal-main_ui-root-active_tab` = "adttespec_example")
+  app$click("teal-main_ui-root-adttespec_example-genes-select_none_button")
+  app$click("teal-main_ui-root-adttespec_example-genes-select_all_button")
+  app$click("teal-main_ui-root-adttespec_example-genes-text_button")
+  app$click("teal-main_ui-filter_panel-remove_all_filters")
+  app$click("teal-main_ui-filter_panel-ADTTE_filter-remove_filters")
+  app$click("teal-main_ui-filter_panel-MAE_filter-remove_filters")
+  app$set_inputs(`teal-main_ui-root-adttespec_example-experiment-name` = "hd1")
+  app$set_inputs(`teal-main_ui-root-adttespec_example-assay-name` = "")
+  app$set_inputs(`teal-main_ui-root-adttespec_example-genes-genes` = character(0))
+  app$set_inputs(`teal-main_ui-root-adttespec_example-genes-fun_name` = "Mean")
+  app$set_inputs(`teal-main_ui-root-adttespec_example-adtte-paramcd` = "")
+  app$set_inputs(`teal-main_ui-root-adttespec_example-genes-lock_button` = FALSE)
+  app$expect_values()
+  app$set_inputs(`teal-main_ui-root-adttespec_example-genes-genes` = "GeneID:28")
+
+  # Upon initialization the endpoint is not selected automatically, the user
+  # has to click this actively.
+  app$expect_values()
```

Comment:
I think in this test file there is too many blanket snapshot expectations. I would prefer more specific comparisons, see e.g. https://rstudio.github.io/shinytest2/articles/robust.html#example, so first getting specific values and then comparing those